java/pypowsybl/src/main/java/com/powsybl/python/grid2op/Backend.java

```diff
@@ -88,21 +90,28 @@ public class Backend implements Closeable {
     private final int[] branchBusGlobalNum1;
     private final int[] branchBusGlobalNum2;
 
+    record TopoChange(Terminal terminal, String newBusId, boolean connected) {
+    }
+
     private int[] loadTopoVectPosition;
     private int[] generatorTopoVectPosition;
     private int[] branchTopoVectPosition1;
     private int[] branchTopoVectPosition2;
     private ArrayPointer<CIntPointer> topoVect;
 
+    private final List<TopoChange> topoChanges = new ArrayList<>();
+
     private final LoadFlowProvider loadFlowProvider = LoadFlowProvider.findAll().stream()
             .filter(p -> p.getName().equals("OpenLoadFlow"))
             .findFirst()
             .orElseThrow();
     private final LoadFlow.Runner loadFlowRunner = new LoadFlow.Runner(loadFlowProvider);
 
-    public Backend(Network network, boolean considerOpenBranchReactiveFlow, int busesPerVoltageLevel, boolean connectAllElementsToFirstBus) {
+    public Backend(Network network, boolean considerOpenBranchReactiveFlow, boolean checkIsolatedAndDisconnectedInjections,
```

Comment:
Not sure i'm entirely following everything happening here, but I think I have a global understanding of what is happening.

Plus there is a test to make sure the behaviour is working as intended.